### PR TITLE
 New case of raw_image_create_and_check

### PR DESCRIPTION
**Pull request by hellohellenmao**
1. Create and write data to a image
2. Check the virtual and actual size info of the image

ID: 1766421
Signed-off-by: Tingting Mao <timao@redhat.com>

### CODE REVIEW

**Comment by hellohellenmao** [comment]
(1/1) Host_RHEL.m8.u1.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.raw_image_create_and_check: PASS (2.71 s)

**Comment by hellohellenmao** [comment]
@lolyu @YongxueHong  Help to take a review, please.

File: qemu/tests/raw_image_create_and_check.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    def _qemu_io(img, cmd):
+        """Run qemu-io cmd to a given img."""
+        logging.info("Run qemu-io %s" % img.image_filename)
+        try:
+            QemuIOSystem(test, params, img.image_filename).cmd_output(cmd, 120)
+        except process.CmdError:
+            test.fail("qemu-io to '%s' failed." % img.image_filename)
```
Comment (YongxueHong) [inline qemu/tests/raw_image_create_and_check.py]: Does need to use `try ... except` ? Why?

File: qemu/tests/raw_image_create_and_check.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    def _check_img_size(img_info, defined_size, size_key):
+        """Check the size info of the image"""
+        logging.info("Check the '%s' size info of %s"
+                     % (size_key, source.image_filename))
+        defined_size = normalize_data_size(defined_size, "B")
+        get_size = img_info[size_key]
+        if int(defined_size) != int(get_size):
+            test.fail("Got unexpected size '%s', expected size is '%s'"
+                      % (get_size, defined_size))
```
Comment (YongxueHong) [inline qemu/tests/raw_image_create_and_check.py]: I thought you could use` for ... in `and `zip `to deal with the `actual-size` and `virtual-size `at the same time.

**Comment by hellohellenmao** [comment]
(1/1) Host_RHEL.m8.u1.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.raw_image_create_and_check: PASS (2.71 s)

File: qemu/tests/raw_image_create_and_check.py (hunk 2 above)
Comment (hellohellenmao) [reply qemu/tests/raw_image_create_and_check.py]: Updated, thanks Yongxue.

**Comment by YongxueHong** [comment]
>     1. Create and write data to a image
> 
>     2. Check the virtual and actual size info of the image
> 
> 
> Signed-off-by: Tingting Mao [timao@redhat.com](mailto:timao@redhat.com)

Please add the related bz id.
For example:
```
2. Check the virtual and actual size info of the image

ID: xxxxxxxx
Signed-off-by: Tingting Mao timao@redhat.com
```
Thanks.

**Comment by hellohellenmao** [comment]
@YongxueHong done, thanks.

**Comment by hellohellenmao** [comment]
@lolyu Help to review, please.

**Comment by lolyu** [comment]
thanks all.

**Pull request merged**